### ISSUE TITLE
wrong-type-argument error in uninitialized repo

### ISSUE THREAD

**Issue by 501st-alpha1** [jtatarik/magit-gitflow#19]
I recently added a git-flow-enabled repo to a new computer, where I continued work on a feature branch that I had started elsewhere.  When I went to finish the feature (<kbd>C-f f f f</kbd>), I got this error:

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  compare-strings(nil 0 0 "feature/foo" 0 0 nil)
  string-prefix-p(nil "feature/foo")
  (if (string-prefix-p prefix current-branch) (substring current-branch (length prefix)) "")
  (let* ((prefix (magit-get "gitflow.prefix.feature")) (current-branch (magit-get-current-branch)) (current-feature (if (string-prefix-p prefix current-branch) (substring current-branch (length prefix)) ""))) (magit-run-gitflow "feature" args magit-current-popup-args (string-remove-prefix prefix (magit-completing-read "Feature name" (magit-list-refnames "refs/heads") (function (lambda (ref) (string-prefix-p prefix ref))) t current-branch))))
  magit-run-gitflow-feature("finish")
  magit-gitflow-feature-finish()
  funcall-interactively(magit-gitflow-feature-finish)
  #<subr call-interactively>(magit-gitflow-feature-finish nil nil)
  ad-Advice-call-interactively(#<subr call-interactively> magit-gitflow-feature-finish)
  apply(ad-Advice-call-interactively #<subr call-interactively> magit-gitflow-feature-finish)
  call-interactively(magit-gitflow-feature-finish)
  (let* ((magit-current-popup magit-this-popup) (magit-current-popup-args (magit-popup-get-args)) (command (progn nil (or (progn nil (and (vectorp ...) (>= ... 7) (memq ... cl-struct-magit-popup-event-tags) t)) (signal (quote wrong-type-argument) (list (quote magit-popup-event) (or action variable)))) (aref (or action variable) 4))) (magit-current-popup-action command)) (if action (progn (magit-popup-quit))) (call-interactively command) (setq this-command command) (if action nil (magit-refresh-popup-buffer)))
  (if (or action variable) (let* ((magit-current-popup magit-this-popup) (magit-current-popup-args (magit-popup-get-args)) (command (progn nil (or (progn nil (and ... ... ... t)) (signal (quote wrong-type-argument) (list ... ...))) (aref (or action variable) 4))) (magit-current-popup-action command)) (if action (progn (magit-popup-quit))) (call-interactively command) (setq this-command command) (if action nil (magit-refresh-popup-buffer))) (if (eq event 113) (progn (magit-popup-quit) (if magit-previous-popup (progn (magit-popup-mode-setup magit-previous-popup nil)))) (user-error "%c isn't bound to any action" event)))
  (let ((action (magit-popup-lookup event :actions)) (variable (magit-popup-lookup event :variables))) (if (and variable (not (progn nil (or (progn nil (and ... ... ... t)) (signal (quote wrong-type-argument) (list ... variable))) (aref variable 3)))) (progn (setq action variable) (setq variable nil))) (if (or action variable) (let* ((magit-current-popup magit-this-popup) (magit-current-popup-args (magit-popup-get-args)) (command (progn nil (or (progn nil ...) (signal ... ...)) (aref (or action variable) 4))) (magit-current-popup-action command)) (if action (progn (magit-popup-quit))) (call-interactively command) (setq this-command command) (if action nil (magit-refresh-popup-buffer))) (if (eq event 113) (progn (magit-popup-quit) (if magit-previous-popup (progn (magit-popup-mode-setup magit-previous-popup nil)))) (user-error "%c isn't bound to any action" event))))
  magit-invoke-popup-action(102)
  funcall-interactively(magit-invoke-popup-action 102)
  #<subr call-interactively>(magit-invoke-popup-action nil nil)
  ad-Advice-call-interactively(#<subr call-interactively> magit-invoke-popup-action nil nil)
  apply(ad-Advice-call-interactively #<subr call-interactively> (magit-invoke-popup-action nil nil))
  call-interactively(magit-invoke-popup-action nil nil)
  command-execute(magit-invoke-popup-action)
```

Eventually I realized that I had not yet run `git flow init` on my new copy of the repo.

I realize this is a bit of an edge case, but it would be nice to have a friendlier error message here.

**Comment by jtatarik** [comment]
I have just tried to execute flow feature start in a plain git repo and I got this nice error message in the minibuffer:

> Not a gitflow-enabled repo yet. Please run 'git flow init' first. ... [Hit $ to see buffer *magit-process: gittest for details]

I think this has to do with the `magit.el` version installed on your system - is it recent enough? What version of `gitflow` and `magit.el` do you have installed?

**Comment by 501st-alpha1** [comment]
> I have just tried to execute flow feature start ...

I am actually seeing this error when running `flow feature finish` (<kbd>C-f f f f</kbd>).  I have seen the error you mentioned when trying to start a feature, but the error I got was because I tried to finish a feature that I had started elsewhere.

I'm not at the computer that I got this error on at the moment, but I believe it was a fresh clone of Magit and magit-gitflow from GitHub.
